src/fropin.cc

```diff
@@ -29,13 +29,13 @@ using libsemigroups::detail::Timer;
 
 // Macros for the GAP version of the algorithm
 
+#define ELM_PLIST2(plist, i, j) ELM_PLIST(ELM_PLIST(plist, i), j)
+
 #define INT_PLIST(plist, i) INT_INTOBJ(ELM_PLIST(plist, i))
 #define INT_PLIST2(plist, i, j) INT_INTOBJ(ELM_PLIST2(plist, i, j))
 
 inline void SET_ELM_PLIST2(Obj plist, UInt i, UInt j, Obj val) {
-  SET_ELM_PLIST(ELM_PLIST(plist, i), j, val);
-  SET_LEN_PLIST(ELM_PLIST(plist, i), j);
-  CHANGED_BAG(plist);
```

Comment:
This should have been `CHANGED_BAG(ELM_PLIST(plist, i));`, so in the above form there is potential for a GC crash. But of course `ASS_LIST` also does that for us now.